sycl/source/detail/pi.cpp

```diff
@@ -375,6 +375,12 @@ const std::vector<plugin> &initialize() {
     initializePlugins(&GlobalHandler::instance().getPlugins());
   });
 
+  // reset LastDeviceIds to zeros
+  vector_class<plugin> Plugins = GlobalHandler::instance().getPlugins();
+  for (plugin Plugin : Plugins) {
+    Plugin.resetLastDeviceIds();
+  }
```

Comment:
Plugins are initialized only once because it is wrapped with std::call_once.
If a thread tries to get plugins before they are initialized, getPlugins will return an empty vector, but I don't think any codes is doing that in SYCL RT.